services/src/main/java/org/fao/geonet/api/records/formatters/XsltFormatter.java

```diff
@@ -97,16 +115,33 @@ public String format(FormatterParams fparams) throws Exception {
 
         root.addContent(new Element("lang").setText(fparams.context.getLanguage()));
         root.addContent(new Element("url").setText(fparams.url));
-        // FIXME: This is a hack to mimic what Jeeves service are doing.
-        // Some XSLT are used by both formatters and Jeeves and Spring MVC services
-        Element translations = new Element("translations");
+
+        if (!translationElements.containsKey(fparams.context.getLanguage())) {
+            // Get translation keys and add them to the cache
+            Element translations = new Element("translations");
+            Map<String, String> translationMap = new JSONLocCacheLoader(configurableApplicationContext, fparams.context.getLanguage()).call();
+            for (Map.Entry<String, String> entry : translationMap.entrySet()) {
+                // Skip keys that are not alpha numeric only including "." - otherwhise chars like : ? +... cause problem when creating element.
```

Comment:
Can you indicate some translation including such values in the keys?

Reply:
Updated comments to show some sample keys that cause issues.